src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1651,5 +1645,68 @@ private void ValidateOrganizationUpgradeParameters(Models.StaticStore.Plan plan,
                     $"{plan.MaxAdditionalSeats.GetValueOrDefault(0)} additional users.");
             }
         }
+
+        private async Task ValidateOrganizationUserUpdatePermissions(Guid loggedInUserId, Guid organizationId, OrganizationUserType newType, OrganizationUserType? oldType) 
+        {
+            var checkedTypes = new List<OrganizationUserType> 
+            { 
+                OrganizationUserType.Owner,
+                OrganizationUserType.Admin,
+                OrganizationUserType.Custom 
+            };
+
+            if (checkedTypes.Contains(newType) || (oldType.HasValue && checkedTypes.Contains(oldType.Value)))
+            {
+                var loggedInUserOrgs = await _organizationUserRepository.GetManyByUserAsync(loggedInUserId);
+                var loggedInAsOrgOwner = loggedInUserOrgs
+                    .Any(u => u.OrganizationId == organizationId && u.Type == OrganizationUserType.Owner);
+                if (!loggedInAsOrgOwner)
+                {
+                    var isOwner = oldType.HasValue ?
+                        oldType.Value == OrganizationUserType.Owner :
+                        false;
+                    var nowOwner = newType == OrganizationUserType.Owner;
+                    var ownerUserConfigurationAttempt = (isOwner && nowOwner) || !(isOwner.Equals(nowOwner));
+                    if (ownerUserConfigurationAttempt)
+                    {
+                        throw new BadRequestException("Only an Owner can configure another Owner's account.");
+                    }
+
+                    var loggedInAsOrgAdmin = loggedInUserOrgs.Any(u => u.OrganizationId == organizationId && u.Type == OrganizationUserType.Admin);
+                    if (!loggedInAsOrgAdmin)
+                    {
+                        var isCustom = oldType.HasValue ? 
+                           oldType.Value == OrganizationUserType.Custom :
+                           false;
+                        var nowCustom = newType == OrganizationUserType.Custom;
+                        var customUserConfigurationAttempt = (isCustom && nowCustom) || !(isCustom.Equals(nowCustom));
+                        if (customUserConfigurationAttempt)
+                        {
+                            throw new BadRequestException("Only Owners and Admins can configure Custom accounts.");
+                        }
+
+                        var loggedInAsOrgCustomOrgUser = loggedInUserOrgs.FirstOrDefault(u => u.OrganizationId == organizationId && u.Type == OrganizationUserType.Custom);
+                        if (loggedInAsOrgCustomOrgUser != null)
+                        {
+                            if (!loggedInAsOrgCustomOrgUser.ManageUsers)
+                            {
+                                throw new BadRequestException("Your account is not permissioned to manage users.");
```

Comment:
please consider rewording, `"Your account does not have permission to manage users"`